modules/planning/navi/decider/navi_obstacle_decider.cc

```diff
@@ -152,50 +178,121 @@ int NaviObstacleDecider::ProcessPathObstacle(
           current_obstacle->Perception().width(), dist));
     }
   }
-  return obstacles_num;
+}  // namespace planning
+
+double NaviObstacleDecider::ReturnReferenceLineJudgment(
+    std::map<double, double>::iterator iter, const double right_nudge_lane,
+    const double left_nudge_lane, int* lane_obstacles_num) {
+  auto obs_width = iter->first;
+  auto lat_dist = iter->second;
+  ADEBUG << "get obstacle width : " << obs_width
+         << "get lattitude distance : " << lat_dist;
+  auto actual_dist =
+      fabs(lat_dist) - obs_width / 2.0 - VehicleParam().left_edge_to_center();
+  if (last_nudge_dist_ > 0) {
+    if (lat_dist < 0) {
+      if (actual_dist < fabs(right_nudge_lane) + config_.safe_distance()) {
+        *lane_obstacles_num = *lane_obstacles_num + 1;
+      }
+    }
+  } else if (last_nudge_dist_ < 0) {
+    if (lat_dist > 0) {
+      if (actual_dist < fabs(left_nudge_lane) + config_.safe_distance()) {
+        *lane_obstacles_num = *lane_obstacles_num + 1;
+      }
+    }
+  }
+  ADEBUG << "last_nudge_dist : " << last_nudge_dist_
+         << "lat_dist : " << lat_dist << "actual_dist : " << actual_dist;
+  return actual_dist;
+}
+
+void NaviObstacleDecider::RecordLastNudgeDistance(const double nudge_dist) {
+  static unsigned int no_nudge_num = 0;
```

Comment:
need to reset static variables when starting (or stop) the planning module.

Reply:
Yes, it should be.